docs/enhancements/ova-upload.md

```diff
@@ -0,0 +1,169 @@
+
+Due to this increase in complexity, the provider server should be refactored to use a more feature rich server
+(preferably Gin which is used in Forklift for the inventory) and reorganized into multiple files. Special consideration
+needs to be given to the upload endpoint to ensure that uploaded appliances are streamed to disk efficiently. Configuration
+options should be provided to set the maximum accepted file size.
+
+### Security, Risks, and Mitigations
+
+The upload component of this enhancement permits uploading and deleting arbitrary files from a volume in the cluster,
+and those arbitrary uploads will be used in migration to construct VMs that will run in the cluster. This is an inherently
+risky operation that is acceptable given the risk already inherent in migrating VMs into the cluster.
+
+The provider server endpoints should be protected by auth, as is done for the inventory endpoints, and a configuration 
+option should be provided so that the administrator can disable the OVA upload/delete endpoints.
+
+## Design Details
+
+### Test Plan
+
+Integration tests should verify that the OVA provider server deployment is reconfigured when the OVA provider's
+storage configuration is changed, that providers configured with NFS shares continue to function, that
+the catalogue management endpoints work, and that uploaded OVAs can be imported.
+
+### Upgrade / Downgrade Strategy
+
+OVA providers that rely on NFS shares can continue to be supported until it becomes reasonable
+to deprecate and remove direct consumption of NFS shares. The provider controller shall update
+any existing OVA provider server deployments to ensure they refer to the correct version of
+the image.
+
+An OVA Provider resource that has been created with a reference to a PVC instead of an NFS share
+will not be able to be inventoried or used in a plan if Forklift is downgraded to a prior version. The existing
+version of Forklift does not update the OVA provider server deployment if the provider is changed, so
+downgrading will require any OVA provider server deployments to be removed so that the provider controller
+can recreate them with the correct version of the image.
+
+## Implementation History
+
+* 1/13/2025: Enhancement pull request opened
+
+## Drawbacks
+
+Allowing the end user to provide a PVC of their choice to use for the appliance catalogue rather than requiring an
+NFS share is a purely beneficial improvement in usability and aligns better with Kube/OpenShift patterns.
+Likewise, allowing upload of images into the store is a significant improvement in usability that comes at the minor
+cost of a slightly more complex provider server to handle the file upload. There are no meaningful drawbacks to these
+usability improvements.
+
+## Alternatives
+
+### Inventorying remote appliances
+
+An alternative to storing appliances locally is to store the URLs of appliances, and then use nbdkit to remotely examine
```

Comment:
yeah nbdkit would be another alternative, do you think these features go against each other?
In my mind we could do both features, depending on the time.
The problem with the nbdkit is that the customer would still need to setup some HTTP server, where as what this enhancement describes would allow users to easily upload directly.

Reply:
I don't think the two approaches are necessarily mutually exclusive; both could be supported. But I believe that remotely inventorying appliances requires larger design changes that would be better approached in a follow up feature.